TUFLOW HPC linked models in LF1

scleakey opened the pull request:
- Added python 3.12 to testing pipeline, which involved setting a higher version to lxml
- Formatted files according to newer version of black
- Added extra LF1 params for TUFLOW HPC linked models
- Amended LF functionality so it works for time series that don't appear in log file

scleakey commented:
(closes #100)

joe-pierce commented:
Thanks Shannon - before I merge can you add something extra to the docs to show this in an example, and update the docstrings that are used in the docs: https://api.floodmodeller.com/api/user_guide/lf.html

scleakey commented:
@joe-pierce I updated the documentation as requested. I also added sphinx to the contributing guidelines and requirements.txt, which required updating from sphinx-panels to sphinx-design. I also added sphinx to github actions as it was picking up a few important docstring formatting errors.
